src/view/observer/focusobserver.js

```diff
@@ -22,8 +24,21 @@ export default class FocusObserver extends DomEventObserver {
 		this.domEventType = [ 'focus', 'blur' ];
 	}
 
-	onDomEvent( domEvt ) {
-		this.fire( domEvt.type, domEvt );
+	onDomEvent( domEvent ) {
+		const target = this.document.domConverter.getCorrespondingViewElement( domEvent.target );
+
+		// Update `isFocus` property of root elements.
+		for ( let root of this.document.roots.values() ) {
```

Comment:
Well... it could. But then we would have to take case of unbinding this event and care more about them. Also it is fine for observer to extend view. For instance `SelectionObserver` flush mutations. `MutationObserver` mark elements to be rerenders. The idea of observers was more like the strategy pattern, adding some logic to the view layer, then only firing events.